mobile/src/bundles/talent/components/profile-form/profile-form.tsx

```diff
@@ -10,27 +11,57 @@ import {
     Slider,
     View,
 } from '~/bundles/common/components/components';
+import { splitArrayInHalf } from '~/bundles/common/helpers/helpers';
 import { useAppForm, useCallback } from '~/bundles/common/hooks/hooks';
 import { globalStyles } from '~/bundles/common/styles/styles';
+import { type TalentOnboardingProfileDto } from '~/bundles/talent/types/types';
 
-import {
-    CURRENT_LOCATION_OPTIONS,
-    JOB_TITLE_OPTIONS,
-    USER_PROFILE_DEFAULT_VALUES,
-} from './constants/constants';
+import { TALENT_PROFILE_DEFAULT_VALUES } from './constants/constants';
 import { styles } from './styles';
 
+const jobTitleOptions = Object.values(JobTitle).map((title) => ({
+    value: title,
+    label: title,
+}));
+
+const locationOptions = Object.values(CountryList).map((country) => ({
+    value: country,
+    label: country,
+}));
+
+const employmentTypeOptions = Object.values(EmploymentType).map((type) => ({
+    value: type,
+    label: type,
+}));
+
+const [column1Options, column2Options] = splitArrayInHalf(
+    employmentTypeOptions,
+);
+
+const extractSelectedEmploymentTypes = (data: string[]): string[] => {
+    return employmentTypeOptions
+        .filter((_, index) => data[index])
+        .map((option) => option.value);
+};
+
 type Properties = {
-    onSubmit: () => void;
+    onSubmit: (payload: TalentOnboardingProfileDto) => void;
 };
 
 const ProfileForm: React.FC<Properties> = ({ onSubmit }) => {
     const { control, errors, handleSubmit } = useAppForm({
-        defaultValues: USER_PROFILE_DEFAULT_VALUES,
+        defaultValues: TALENT_PROFILE_DEFAULT_VALUES,
     });
 
-    const handleFormSubmit = useCallback((): void => {
-        void handleSubmit(onSubmit)();
+    const handleFormSubmit = useCallback(() => {
+        void handleSubmit((data) => {
+            data.salaryExpectation = Number(data.salaryExpectation);
+            data.employmentTypes = extractSelectedEmploymentTypes(
+                data.employmentTypes,
+            );
+
```

Comment:
Since employmentTypes should be an array, I get the following value from the form: 
`"employmentTypes": [true, undefined, true, true, undefined]`

And I transform it into the following:
`"employmentTypes": ["Full Time", "Freelance/Projects", "Remotely"]`